<?xml version="1.0" encoding="UTF-8"?>
<class xmlns="http://xml.phpdox.net/src" full="Symfony\Component\Filesystem\Exception\IOException" namespace="Symfony\Component\Filesystem\Exception" name="IOException" abstract="false" final="false" start="21" end="39">
  <file path="//var/www/html/vendor/symfony/filesystem/Exception" file="IOException.php" realpath="/var/www/html/vendor/symfony/filesystem/Exception/IOException.php" size="956" time="2022-04-01T12:33:59+00:00" unixtime="1648816439" sha1="d8adb868fbd22ee2c941fcb7679ef712d21886bf" relative="ar/www/html/vendor/symfony/filesystem/Exception/IOException.php"/>
  <docblock>
    <description compact="Exception class thrown when a filesystem operation failure happens."/>
    <author value="Romain Neutron &lt;imprec@gmail.com&gt;"/>
    <author value="Christian GÃ¤rtner &lt;christiangaertner.film@googlemail.com&gt;"/>
    <author value="Fabien Potencier &lt;fabien@symfony.com&gt;"/>
  </docblock>
  <extends full="RuntimeException" namespace="" name="RuntimeException"/>
  <implements full="Symfony\Component\Filesystem\Exception\IOExceptionInterface" namespace="Symfony\Component\Filesystem\Exception" name="IOExceptionInterface"/>
  <member name="path" visibility="private" static="false" line="23"/>
  <constructor name="__construct" start="25" end="30" abstract="false" final="false" static="false" visibility="public">
    <parameter name="message" byreference="false" type="string"/>
    <parameter name="code" byreference="false" type="integer" default="0"/>
    <parameter name="previous" byreference="false" type="object" default="NULL">
      <type full="Throwable" namespace="" name="Throwable"/>
    </parameter>
    <parameter name="path" byreference="false" type="string" default="NULL"/>
  </constructor>
  <method name="getPath" start="35" end="38" abstract="false" final="false" static="false" visibility="public">
    <docblock>
      <description compact="{@inheritdoc}"/>
    </docblock>
  </method>
  <parent full="RuntimeException" namespace="" name="RuntimeException">
    <extends full="Exception" namespace="" name="Exception"/>
    <member name="message" static="false" visibility="protected">
    <default>null</default>
    <docblock>
      <description compact="The exception message"/>
      <var type="string"/>
    </docblock>
  </member>
    <member name="code" static="false" visibility="protected">
    <default>null</default>
    <docblock>
      <description compact="The exception code"/>
      <var type="int"/>
    </docblock>
  </member>
    <member name="file" static="false" visibility="protected">
    <default>null</default>
    <docblock>
      <description compact="The filename where the exception was created"/>
      <var type="string"/>
    </docblock>
  </member>
    <member name="line" static="false" visibility="protected">
    <default>null</default>
    <docblock>
      <description compact="The line where the exception was created"/>
      <var type="int"/>
    </docblock>
  </member>
    <method name="getMessage" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact=""/>
      <return type="string"/>
    </docblock>
  </method>
    <method name="getPrevious" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact=""/>
      <return type="Throwable"/>
    </docblock>
  </method>
    <method name="getCode" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact=""/>
      <return type="mixed"/>
    </docblock>
  </method>
    <method name="getFile" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact=""/>
      <return type="string"/>
    </docblock>
  </method>
    <method name="getLine" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact=""/>
      <return type="int"/>
    </docblock>
  </method>
    <method name="getTrace" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact=""/>
      <return type="array"/>
    </docblock>
  </method>
    <method name="getTraceAsString" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact=""/>
      <return type="string"/>
    </docblock>
  </method>
    <method name="__toString" abstract="false" static="false" visibility="public" final="false">
    <docblock>
      <description compact=""/>
      <return type="string"/>
    </docblock>
  </method>
  </parent>
  <parent full="Exception" namespace="" name="Exception">
    <member name="message" static="false" visibility="protected">
    <default>null</default>
    <docblock>
      <description compact="The exception message"/>
      <var type="string"/>
    </docblock>
  </member>
    <member name="code" static="false" visibility="protected">
    <default>null</default>
    <docblock>
      <description compact="The exception code"/>
      <var type="int"/>
    </docblock>
  </member>
    <member name="file" static="false" visibility="protected">
    <default>null</default>
    <docblock>
      <description compact="The filename where the exception was created"/>
      <var type="string"/>
    </docblock>
  </member>
    <member name="line" static="false" visibility="protected">
    <default>null</default>
    <docblock>
      <description compact="The line where the exception was created"/>
      <var type="int"/>
    </docblock>
  </member>
    <constructor name="__construct" abstract="false" static="false" visibility="public" final="false">
    <docblock>
      <description compact="Construct the exception"/>
      <return type="void"/>
    </docblock>
    <parameter name="message" optional="true" byreference="false" type="string"/>
    <parameter name="code" optional="true" byreference="false" type="int"/>
    <parameter name="previous" optional="true" byreference="false" type="object" class="Throwable"/>
  </constructor>
    <method name="getMessage" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact="Gets the Exception message"/>
      <return type="string"/>
    </docblock>
  </method>
    <method name="getPrevious" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact="Returns previous Exception"/>
      <return type="Throwable"/>
    </docblock>
  </method>
    <method name="getCode" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact="Gets the Exception code"/>
      <return type="mixed"/>
    </docblock>
  </method>
    <method name="getFile" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact="Gets the file in which the exception was created"/>
      <return type="string"/>
    </docblock>
  </method>
    <method name="getLine" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact="Gets the line in which the exception was created"/>
      <return type="int"/>
    </docblock>
  </method>
    <method name="getTrace" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact="Gets the stack trace"/>
      <return type="array"/>
    </docblock>
  </method>
    <method name="getTraceAsString" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact="Gets the stack trace as a string"/>
      <return type="string"/>
    </docblock>
  </method>
    <method name="__toString" abstract="false" static="false" visibility="public" final="false">
    <docblock>
      <description compact="String representation of the exception"/>
      <return type="string"/>
    </docblock>
  </method>
  </parent>
  <interface full="Symfony\Component\Filesystem\Exception\IOExceptionInterface" namespace="Symfony\Component\Filesystem\Exception" name="IOExceptionInterface">
    <extends full="Symfony\Component\Filesystem\Exception\ExceptionInterface" namespace="Symfony\Component\Filesystem\Exception" name="ExceptionInterface"/>
    <method name="getPath" start="26" end="26" abstract="false" final="false" static="false" visibility="public">
      <docblock>
        <description compact="Returns the associated path for the exception."/>
        <return type="object">
          <type full="Symfony\Component\Filesystem\Exception\string|null" namespace="Symfony\Component\Filesystem\Exception" name="string|null"/>
        </return>
      </docblock>
    </method>
  </interface>
  <extenders>
    <extender full="Symfony\Component\Filesystem\Exception\FileNotFoundException" namespace="Symfony\Component\Filesystem\Exception" name="FileNotFoundException"/>
  </extenders>
</class>
